cnb_image.go

```diff
@@ -285,6 +312,12 @@ func (i *CNBImageCore) SetOS(osVal string) error {
 	})
 }
 
+func (i *CNBImageCore) SetOSFeatures(osFeatures []string) error {
+	return i.MutateConfigFile(func(c *v1.ConfigFile) {
+		c.OSFeatures = osFeatures
```

Comment:
isn't the expected behaviour is to append the requested `osFeatures` instead of replacing/overriding with existing one?

Reply:
@jjbustamante do you know?

Reply:
Maybe we should call it `AppendOSFeatures`? But also, are we using this method in pack?

Reply:
In the initial requirement we were expecting to set only `os`, `architecture`, `variant` information in a manifest that belongs to an Index. Sai exposed this attribute with the `pack manifest add` and `pack manifest annotate` commands, but I do not think the community is requesting to set these values. Yes, they are in the OCI spec, but so far, nobody is asking for it

Reply:
it seems this is useful for `windows` indexes, I think our plans for windows support are very limited